test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,159 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidSerializer() = default;
+	~TestUuidSerializer() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
+// Test string serialization
+TEST_F(TestUuidSerializer, SerializeToString) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABCDEF);
+	uuid.set_lsb(0xFEDCBA0987654321);
+
+	std::string uuid_str =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	EXPECT_EQ(uuid_str, "1234567890abcdef-fedcba0987654321");
```

Comment:
It wouldn't hurt to add a few extra test values, such as one that might produce leading zeros in each of the segments.